test/Core.Test/Services/LicensingServiceTests.cs

```diff
@@ -38,12 +51,215 @@ public LicensingServiceTests()
             );
         }
 
-        // Remove this test when we add actual tests. It only proves that
-        // we've properly constructed the system under test.
-        [Fact(Skip = "Needs additional work")]
-        public void ServiceExists()
+        [Fact]
+        public async Task ValidateOrganizationsAsync_IsNotSelfHosted_ShouldDoNothing()
+        {
+            // arrange
+            _globalSettings.SelfHosted = false;
+
+            // act
+            await _sut.ValidateOrganizationsAsync();
+
+            // assert
+            await _organizationRepository.DidNotReceive().GetManyByEnabledAsync();
+        }
+
+        [Fact]
+        public async Task ValidateOrganizationAsync_SelfHostedOrgNoLicense_ShouldDisableOrganization()
+        {
+            // Arrange
+            var organizations = new List<Organization>
+            {
+                new Organization
+                {
+                    Id = Guid.NewGuid(),
+                    Enabled = true
+                }
+            };
+            _organizationRepository.GetManyByEnabledAsync().Returns(organizations);
+
+            //Act
+            await _sut.ValidateOrganizationsAsync();
+
+            //Assert
+            await _organizationRepository.Received().ReplaceAsync(organizations.First());
+            await _organizationRepository.Received().ReplaceAsync(Arg.Is<Organization>(o => o.Enabled == false));
+            
+            var loggerArguments = _logger.ReceivedCalls().Last().GetArguments()
+                                                    .Select(arg => arg.ToString())
+                                                    .Where(arg => arg != null)
+                                                    .AsEnumerable<string>();
+            
+            Assert.Contains(loggerArguments, arg => arg.Contains("No license file"));
+        }
+
+        [Fact]
+        public async Task ValidateOrganizationAsync_SelfHostedOrgWithLicenseMultipleOrgsSameKey_ShouldDisableOrg()
+        {
+            // Arrange
+            var organizationId = Guid.NewGuid();
+            var organizations = new List<Organization>
+            {
+                new Organization
+                {
+                    Id = organizationId,
+                    Enabled = true,
+                    LicenseKey = "test"
+                },
+                new Organization
+                {
+                    Id = organizationId,
```

Comment:
If ensuring the test covers a 2 different orgs with the same key, each org should have it's own `Id` to ensure expected behavior would follow those lines (vs. the logic testing/asserting 2 orgs don't exist with the same `Id`, etc.).